### PR TITLE
[Serve] First iteration of the serve doc

### PR DESCRIPTION
**Pull request by simon-mo**

### CODE REVIEW

File: doc/source/serve.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
+Ray Serve (Experimental)
+===========================
```
Comment (richardliaw) [inline doc/source/serve.rst]:
```suggestion
========================
```

File: doc/source/serve.rst [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+Ray serve is a serving library that supports flexible traffic routing,
+version upgrade/downgrade, rolling update and rollback.
```
Comment (richardliaw) [inline doc/source/serve.rst]: Can you provide examples of doing this

File: doc/source/serve.rst [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+===========================
+
+Ray serve is a serving library that supports flexible traffic routing,
+version upgrade/downgrade, rolling update and rollback. 
+
+Quickstart
+----------
+.. code-block:: python
+
+    import ray.experimental.serve as serve
```
Comment (richardliaw) [inline doc/source/serve.rst]: can you put this in a python file, use `literalinclude` and test it on travis?

File: python/ray/experimental/serve/examples/echo_full.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
+"""
+Full example of ray.serve module
+"""
+
+import ray
```
Comment (richardliaw) [inline python/ray/experimental/serve/examples/echo_full.py]: Test this on travis?

File: python/ray/experimental/serve/examples/echo_full.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+# we can link an endpoint to a backend, the means all the traffic
+# goes to my_endpoint will now goes to echo:v1 backend
+serve.link("my_endpoint", "echo:v1")
+
+print(requests.get("http://127.0.0.1:8000/echo").json())
+# the service will be reachable from http
+
+print(ray.get(serve.get_handle("my_endpint").remote()))
+
```
Comment (richardliaw) [inline python/ray/experimental/serve/examples/echo_full.py]:
```suggestion
```

File: doc/source/serve.rst [hunk 6]
```diff
@@ -0,0 +1,17 @@
+Ray Serve (Experimental)
+========================
+
+Ray serve is a serving library that exposes python function/classes to HTTP.
```
Comment (richardliaw) [inline doc/source/serve.rst]: Ray Serve? (everywhere?)

**Comment by simon-mo** [comment]
@richardliaw I messed up rebase somehow. new PR opened at https://github.com/ray-project/ray/pull/5834

**Comment by richardliaw** [comment]
git merge

**Pull request closed without merging**